### DISCUSSION TITLE
More accurate RPM?

### DISCUSSION THREAD

**Discussion by Festivejelly** [kachurovskiy/nanoels#216] [Q&A]
Is there a way to get a more accurate readout of the RPM? It bounces around a little, not a lot by maybe 5 or 10 RPM.

I was thinking this probably will affect threading?

At the moment im using a 600/r encoder, so should be fairly accurate.

I wonder if there is a way to smooth out the RPM which is what I guess the spindle RPM display is doing?

**Comment by kachurovskiy** [comment]
> Is there a way to get a more accurate readout of the RPM? It bounces around a little, not a lot by maybe 5 or 10 RPM.

Why do you think it's inaccurate? Seems plausible that motor speed is not completely constant.

> I was thinking this probably will affect threading?

RPM doesn't affect threading, it's calculated for display purposes.

> I wonder if there is a way to smooth out the RPM which is what I guess the spindle RPM display is doing?

Yes, we're smoothing it out by calculating over 1 encoder turn and then also don't update the displayed value if it's close enough to the currently shown value: https://github.com/kachurovskiy/nanoels/blob/main/h4/h4.ino#L587

**Reply by Festivejelly** [reply]
Thanks Maxim, I figured something like that was the case. It was just bouncing around but Id bet thats because my motor probably isnt exactly spinning at the displayed RPM. I bet the little LCD on the lathe is an average.
